app/assets/javascripts/data_sets/editTable.js.coffee

```diff
@@ -17,148 +17,239 @@ uploadSettings =
   successEntry: (data, textStatus, jqXHR) ->
     window.location = data['displayURL']
 
-setupTable = (cols, data) ->
-  cols.push
-    id: 'del'
-    field: 'del'
-    name: ''
-    width: 0
-    formatter: (row, cell, value, columnDef, dataContext) ->
-      "<i class='fa fa-close slick-delete'></i>"
-
-  options =
-    autoEdit: true
-    editable: true
-    enableCellNavigation: true
-    enableColumnReorder: false
-    forceFitColumns: true
-    rowHeight: 35
-    syncColumnCellResize: true
-
-  for x, i in data
-    data[i]['id'] = i
-
-  view = new Slick.Data.DataView()
-  grid = new Slick.Grid '#slickgrid-container', view, cols, options
-  currId = data.length
-
-  view.setItems data
-
-  grid.onClick.subscribe (e, args) ->
-    cell = grid.getCellFromEvent e
-    if cell.cell == grid.getColumns().length - 1
-      grid.getEditorLock().commitCurrentEdit()
-      grid.resetActiveCell()
-      item = view.getItem cell.row
-      view.deleteItem item.id
-      grid.invalidate()
-
-      if grid.getDataLength() == 0
-        fields = {id: currId}
-        currId += 1
-        for x in cols
-          fields[x.field] = ''
-        view.addItem fields
-
-  view.onRowCountChanged.subscribe (e, args) ->
-    grid.updateRowCount()
-    grid.render()
-
-  view.onRowsChanged.subscribe (e, args) ->
-    grid.invalidateRow args.rows
-    grid.render()
-
-  $('.edit_table_add').click ->
-    if $('.edit_table_save').hasClass 'disabled'
-      return
-
-    grid.getEditorLock().commitCurrentEdit()
-    grid.resetActiveCell()
+Grid = (cols, data, submit) ->
+  # slickgrid objects
+  view = null
+  grid = null
+
+  # objects relevant to popovers
+  validationPop = null
+  validationPopMsg = null
+  deletionPop = null
+  deletionPopTimer = null
+
+  # ID counter for rows
+  currID = 0
+
+  # queue of save and delete row actions to execute on validation
+  actions = []
+
+  initialize = ->
+    # add the delete button to each row
+    cols.push
+      id: 'del'
+      field: 'del'
+      name: ''
+      width: 0
+      formatter: (row, cell, value, columnDef, dataContext) ->
+        "<i class='fa fa-close slick-delete'></i>"
+
+    # slickgrid's grid options
+    options =
+      autoEdit: true
+      editable: true
+      enableCellNavigation: true
+      enableColumnReorder: false
+      forceFitColumns: true
+      rowHeight: 35
+      syncColumnCellResize: true
+
+    # associate IDs with each row in the table
+    for _, i in data
```

Comment:
Do I change it?  This pretty standard in lots of functional languages when you want to ignore a parameter.